mypy/semanal.py

```diff
@@ -1339,6 +1339,13 @@ def analyze_metaclass(self, defn: ClassDef) -> None:
             assert isinstance(inst, Instance)
             defn.info.declared_metaclass = inst
         defn.info.metaclass_type = defn.info.calculate_metaclass_type()
+        if any(info.is_protocol for info in defn.info.mro):
+            if (not defn.info.metaclass_type or
+                    defn.info.metaclass_type.type.fullname() == 'builtins.type'):
```

Comment:
Maybe add a TODO about checking metaclass conflicts If there is another metaclass.